sympy/polys/matrices/ddm.py

```diff
@@ -101,13 +101,18 @@ class DDM(list):
     is_DDM = True
 
     def __init__(self, rowslist, shape, domain):
+        if not (isinstance(rowslist, list) and all(type(row) is list for row in rowslist)):
```

Comment:
If the number of columns is 1 can it just be a list as for matrix (instead of a list of singleton lists)?

Reply:
I prefer not to add those kinds of complications at this level. The constructor here is not really expected to be something that would be used directly by users and so it is better to be strict about what is expected for inputs: prioritise speed and simplicity of implementation over convenience of use.

The convenience of use way leads to `_handle_creation_inputs`:
https://github.com/sympy/sympy/blob/b7a7ee1b7096486d8ecd460332506e04033cce13/sympy/matrices/matrices.py#L896-L1118
I hope I never have to touch that function again!

Reply:
Oh, btw if you are looking to create a DDM conveniently then it is usually better to convert from something else e.g.:
```python
In [3]: M = Matrix([[1, 2], [3, 4]])

In [4]: M.to_DM() # DomainMatrix
Out[4]: DomainMatrix({0: {0: 1, 1: 2}, 1: {0: 3, 1: 4}}, (2, 2), ZZ)

In [5]: M.to_DM().to_ddm() # DDM
Out[5]: DDM([[mpz(1), mpz(2)], [mpz(3), mpz(4)]], (2, 2), ZZ)

In [6]: M.to_DM().to_sdm() # SDM
Out[6]: SDM({0: {0: mpz(1), 1: mpz(2)}, 1: {0: mpz(3), 1: mpz(4)}}, (2, 2), ZZ)

In [7]: M.to_DM().to_dfm() # DFM (needs ground types = flint)
...
NotImplementedError: DFM requires GROUND_TYPES=flint.
```